informer/integration/custom_resource.go

```diff
@@ -0,0 +1,101 @@
+// +build k8srequired
+
+package integration
+
+import (
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+const (
+	namespace = "informer-integration-test"
+)
+
+type CustomResource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+	Spec              CustomResourceSpec `json:"spec"`
+}
+
+func (t *CustomResource) DeepCopyObject() runtime.Object {
+	return &CustomResource{
+		TypeMeta:   t.TypeMeta,
+		ObjectMeta: *t.ObjectMeta.DeepCopy(),
+		Spec:       t.Spec,
+	}
+}
+
+type CustomResourceSpec struct {
+	ID string `json:"id" yaml:"id"`
+}
+
+type CustomResourceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+	Items           []CustomResource `json:"items"`
+}
+
+func (t *CustomResourceList) DeepCopyObject() runtime.Object {
+	itemsCopy := make([]CustomResource, len(t.Items))
+	for i, item := range t.Items {
+		itemsCopy[i] = item
+	}
+
+	return &CustomResourceList{
+		TypeMeta: t.TypeMeta,
+		ListMeta: *t.ListMeta.DeepCopy(),
+		Items:    itemsCopy,
+	}
+}
+
+func createCustomResource(ID string) error {
```

Comment:
It doesn't create a custom resource. It's a ConfigMap.

Reply:
I remember you improved this to use the config maps at some point. Cleaned up now.

Reply:
Yes, I most likely forgot to delete the old struct, and rename few other things ;)